rest/src/config-create.cpp

```diff
@@ -0,0 +1,82 @@
+#include "config-create.h"
+#include "commands.h"
+#include "discovery-config.h"
+#include <fty/rest/component.h>
+
+namespace fty::disco {
+
+unsigned ConfigCreate::run()
+{
+    rest::User user(m_request);
+    if (auto ret = checkPermissions(user.profile(), m_permissions); !ret) {
+        throw rest::Error(ret.error());
+    }
+
+    commands::config::In in;
+    if (auto ret = pack::json::deserialize(m_request.body(), in); !ret) {
+        throw rest::errors::Internal(ret.error());
+    }
+
+    ConfigDiscovery cd;
+    if (auto ret = cd.load(); !ret) {
+        throw rest::errors::Internal(ret.error());
+    }
+
+    if (in.type.hasValue()) {
+        std::stringstream                ss(in.type.value());
+        ConfigDiscovery::Discovery::Type tmpType;
+        ss >> tmpType;
+        cd.discovery.type = tmpType;
```

Comment:
Why you did it explicitly? It cannot be handled by pack?

Reply:
no, it is not handled

Reply:
But why?